aten/src/ATen/native/BatchLinearAlgebra.h

```diff
@@ -239,5 +241,14 @@ using lu_solve_trans_fn = void (*)(
     TransposeType /*trans*/);
 DECLARE_DISPATCH(lu_solve_trans_fn, lu_solve_trans_stub);
 
+using svd_fn = void (*)(
+    const Tensor& /*A*/,
+    const bool /*full_matrices*/,
+    const bool /*compute_uv*/,
+    const Tensor& /*U*/,
+    const Tensor& /*S*/,
+    Tensor& /*Vh*/,
```

Comment:
In the cusolver case we need to conjugate-transpose it in place.

Reply:
In-place operations do not require the Tensor to be non-const. It's different to have a const reference to Tensor (that I propose) and to have non-modifiable elements of Tensor (like `vector<const T>` vs. `const vector<T>`).

Reply:
Note that `vector<const T>` and `const vector<T>` are equivalent because `vector` is const-correct. Now, I see what you mean for `Tensor&` as it uses the PImpl idiom, but I still think that it will not work. I'm going to try it though see what happens.

Reply:
You were right, it does compile. Changing it to const.